flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/slotmanager/DefaultResourceAllocationStrategyTest.java

```diff
@@ -138,6 +137,99 @@ void testFulfillRequirementWithRegisteredResourcesEvenly() {
+                .hasSize(1);
+
+        assertThat(result.getAllocationsOnRegisteredResources().get(jobId).values())
+                .overridingErrorMessage("Expected all 6 allocations on registered TM.")
+                .allSatisfy(
+                        resourceCounter ->
+                                assertThat(resourceCounter.getTotalResourceCount()).isEqualTo(6));
+
+        assertThat(result.getAllocationsOnRegisteredResources().get(jobId).values())
+                .allSatisfy(
+                        resourceCounter ->
+                                assertThat(resourceCounter.containsResource(largeResource))
+                                        .isTrue());
+    }
+
+    @Test
+    void testFullFillRequirementWithUtilizedRegisteredResourceIsPicked() {
+        final TaskManagerInfo utilizedTaskManager =
+                new TestingTaskManagerInfo(
+                        DEFAULT_SLOT_RESOURCE.multiply(10),
+                        DEFAULT_SLOT_RESOURCE.multiply(5),
+                        DEFAULT_SLOT_RESOURCE);
+
+        final TaskManagerInfo freeTaskManager =
+                new TestingTaskManagerInfo(
+                        DEFAULT_SLOT_RESOURCE.multiply(10),
+                        DEFAULT_SLOT_RESOURCE.multiply(10),
+                        DEFAULT_SLOT_RESOURCE);
+        final JobID jobId = new JobID();
+        final List<ResourceRequirement> requirements = new ArrayList<>();
+        final ResourceProfile largeResource = DEFAULT_SLOT_RESOURCE.multiply(2);
+        final TaskManagerResourceInfoProvider taskManagerResourceInfoProvider =
+                TestingTaskManagerResourceInfoProvider.newBuilder()
+                        .setRegisteredTaskManagersSupplier(
+                                () -> Arrays.asList(freeTaskManager, utilizedTaskManager))
+                        .build();
+        requirements.add(ResourceRequirement.create(largeResource, 2));
+        requirements.add(ResourceRequirement.create(ResourceProfile.UNKNOWN, 1));
+
+        final ResourceAllocationResult result =
+                MIN_RESOURCES_STRATEGY.tryFulfillRequirements(
+                        Collections.singletonMap(jobId, requirements),
+                        taskManagerResourceInfoProvider,
+                        resourceID -> false);
+
+        assertAllSlotsAllocatedToRegisteredTaskManagersOnly(result);
+
+        assertThat(result.getAllocationsOnRegisteredResources().get(jobId).keySet())
+                .overridingErrorMessage(
+                        "Expected requirements to be fulfilled by the utilized task manager.")
+                .containsExactly(utilizedTaskManager.getInstanceId());
+
+        assertThat(result.getAllocationsOnRegisteredResources().get(jobId).values())
+                .allSatisfy(
+                        resourceCounter ->
+                                assertThat(resourceCounter.containsResource(largeResource))
+                                        .isTrue());
+    }
```

Comment:
```suggestion
                        resourceCounter -> {
                            assertThat(resourceCounter.containsResource(largeResource))
                                    .isTrue();
                            assertThat(resourceCounter.getTotalResourceCount()).isEqualTo(3);
                        });
```
nit: can we also add the expected resource count?

Reply:
@XComp Thanks, addressed as well ✅